UnityProject/Assets/Scripts/Equipment/Equipment.cs

```diff
@@ -27,11 +29,32 @@ private void Start()
 		playerNetworkActions = gameObject.GetComponent<PlayerNetworkActions>();
 		playerScript = gameObject.GetComponent<PlayerScript>();
 		playerInventory = InventoryManager.AllClientInventorySlots;
-		suitStorageSlot = playerInventory.Find(s => s.SlotName == "suitStorage");
+
+		gasSlots = InitPotentialGasSlots();
 		maskSlot = playerInventory.Find(s => s.SlotName == "mask");
 		idPrefab = Resources.Load<GameObject>("ID");
 	}
 
+	private InventorySlot[] InitPotentialGasSlots()
+	{
+		var playerInventoryDic = new Dictionary<string, InventorySlot>();
+		foreach ( var slot in playerInventory )
+		{
+			playerInventoryDic.Add( slot.SlotName, slot );
+		}
+
+		var slots = new List<InventorySlot>();
+		foreach ( var gasSlot in GasContainer.GasSlots )
+		{
+			if ( playerInventoryDic.ContainsKey( gasSlot ) )
+			{
+				slots.Add( playerInventoryDic[gasSlot] );
+			}
+		}
+
```

Comment:
You can simplify this whole function by replacing it with this:
```suggestion
		var slots = new List<InventorySlot>();
		foreach (var slot in playerInventory)
		{
			if (GasContainer.GasSlots.Contains(slot.SlotName))
			{
				slots.Add(slot);
			}
		}

```

No need to create a new dictionary

Reply:
yeah, good catch